File: artifacts/libopenaptx/others/recode/tool_projects/orbit/libopenaptx/c/tests/test_padding.c
#include "openaptx.h"
#include "test_common.h"
#include <stdint.h>

void test_encode_padding() {
    struct aptx_context *ctx = aptx_init(0);

    uint8_t pcm[3] = {1, 2, 3};  
    uint8_t out[8] = {0};
    size_t written = 0;

    size_t used = aptx_encode(ctx, pcm, sizeof(pcm), out, sizeof(out), &written);
    ASSERT(written == 0, "Should not write on partial input");

    aptx_encode_finish(ctx, out, sizeof(out), &written);
    ASSERT(written > 0, "Finish should pad with silence");

    PASS("aptX encode finish with padding");
    aptx_finish(ctx);
}

File: artifacts/libopenaptx/others/recode/tool_projects/orbit/libopenaptx/rust/tests/test_padding.rs
//! Encoding padding tests
//! Copyright (C) 2018-2021  Pali Rohár <pali.rohar@gmail.com>
//! Rust translation (C) 2026

#[allow(unused)]
mod common;

use libopenaptx::{aptx_init, aptx_finish, aptx_encode, aptx_encode_finish};

#[test]
fn test_encode_padding() {
    let mut ctx = aptx_init(0).expect("Failed to initialize encoder");

    let pcm = [1u8, 2u8, 3u8];
    let mut out = [0u8; 8];
    let mut written: usize = 0;

    let pcm_len = pcm.len();
    let out_len = out.len();
    let used = aptx_encode(&mut ctx, &pcm, pcm_len, &mut out, out_len, &mut written);
    let _ = used;  // Suppress unused variable warning
    assert!(written == 0, "Should not write on partial input");

    aptx_encode_finish(&mut ctx, &mut out, out_len, &mut written);
    assert!(written > 0, "Finish should pad with silence");

    println!("aptX encode finish with padding passed");
    aptx_finish(ctx);
}
